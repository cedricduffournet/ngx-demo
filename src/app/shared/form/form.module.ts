import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { ReactiveFormsModule } from '@angular/forms';

import { TranslateModule } from '@ngx-translate/core';
import { TextInputComponent } from '@app/shared/form/text-input';
import { TextAreaComponent } from '@app/shared/form/text-area';

@NgModule({
  imports: [CommonModule, ReactiveFormsModule, TranslateModule],
  exports: [TextInputComponent, TextAreaComponent],
  declarations: [TextInputComponent, TextAreaComponent]
})
export class FormModule {}
