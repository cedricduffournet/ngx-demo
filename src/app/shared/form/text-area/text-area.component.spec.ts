import { Component } from '@angular/core';
import { ReactiveFormsModule, FormControl, Validators } from '@angular/forms';
import {
  ComponentFixture,
  TestBed,
  fakeAsync,
  async,
  tick
} from '@angular/core/testing';
import { TranslateModule } from '@ngx-translate/core';
import { FormModule } from '@app/shared/form';
import { validateEmail, validatePassword } from '@app/shared/validators';

@Component({
  template: `
    <app-text-area
      [helpText]="helpText"
      [label]="label"
      [showError]="showError"
      [placeholder]="placeholder"
      [errorText]="errorText"
      [formGroupClass]="formGroupClass"
      [formControl]="controlTest"
      [submitted]="submitted"
    ></app-text-area>
  `
})
export class TextInputTestComponent {
  label = '';
  helpText = '';
  showError = false;
  placeholder = '';
  errorText = '';
  formGroupClass = '';
  submitted = false;
  controlTest: FormControl = new FormControl();
}

describe('TextAreaComponent', () => {
  let fixture: ComponentFixture<TextInputTestComponent>;
  let component: TextInputTestComponent;
  let inputElement: any;

  beforeEach(() => {
    TestBed.configureTestingModule({
      imports: [FormModule, ReactiveFormsModule, TranslateModule.forRoot()],
      declarations: [TextInputTestComponent]
    });

    fixture = TestBed.createComponent(TextInputTestComponent);
    component = fixture.componentInstance;
    inputElement = fixture.nativeElement.querySelector('textarea');
  });

  it('should be created', () => {
    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });

  it('should init form controle with text', () => {
    component.controlTest = new FormControl('this is my text');

    fixture.detectChanges();

    expect(inputElement.value).toEqual('this is my text');
  });

  it('should get new value when subscribe to valueChanges', async(() => {
    component.controlTest = new FormControl();

    fixture.detectChanges();
    component.controlTest.valueChanges.subscribe((value: string) => {
      expect(value).toBe('new value');
    });

    inputElement.value = 'new value';
    inputElement.dispatchEvent(new Event('input'));
  }));

  it('input should be disabled', () => {
    component.controlTest = new FormControl({ value: '', disabled: true });

    fixture.detectChanges();

    expect(inputElement.disabled).toBeTruthy();
  });

  it('input should display required error message', () => {
    component.controlTest = new FormControl('', Validators.required);
    component.submitted = true;

    fixture.detectChanges();
    const error = fixture.nativeElement.querySelector('.error-required');
    expect(error).toBeTruthy();
  });


  it('input should display custom error', () => {
    component.showError = true;
    component.errorText = 'Error';

    fixture.detectChanges();
    const error = fixture.nativeElement.querySelector('.custom-error');
    expect(error).toBeTruthy();
  });

  it('input should should have placeholder', () => {
    component.placeholder = 'placeholder';
    fixture.detectChanges();
    expect(inputElement.placeholder).toBe('placeholder');
  });
});
