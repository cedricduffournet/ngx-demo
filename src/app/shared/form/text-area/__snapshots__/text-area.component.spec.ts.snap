// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextAreaComponent should be created 1`] = `
<ng-component
  controlTest={[Function FormControl]}
  errorText=""
  formGroupClass=""
  helpText=""
  label=""
  placeholder=""
  showError="false"
  submitted="false"
>
  <app-text-area
    class="ng-untouched ng-pristine ng-valid"
    ng-reflect-error-text=""
    ng-reflect-form="[object Object]"
    ng-reflect-form-group-class=""
    ng-reflect-help-text=""
    ng-reflect-label=""
    ng-reflect-placeholder=""
    ng-reflect-show-error="false"
    ng-reflect-submitted="false"
  >
    <div
      class="form-group "
    >
      <label
        class="control-label"
      >
        
      </label>
      <textarea
        class="form-control"
        placeholder=""
        rows="5"
      >
        
    
      </textarea>
      
      
      
    </div>
  </app-text-area>
</ng-component>
`;
